src/common/SteelSheets.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+#include "marlin_client.h"
+#include "eeprom.h"
+
+class SteelSheets {
+public:
+    static constexpr float Z_OFFSET_MIN = -2.0F;
+    static constexpr float Z_OFFSET_MAX = 0.0F;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Sets offset for sheet
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @param[in] offset of the sheet
+    /// @return True when successful, false if not
+    static bool SetSheetOffset(uint32_t index, float offset);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Iterate across the profiles and switch to the next calibrated.
+    ///
+    /// Printer use print sheet profile on the index 0 as a default so the method
+    /// in the worst case iterate across entire profiles and return index 0 when
+    /// not any other profile is calibrated yet.
+    /// @return Index of the next calibrated profile.
+    static uint32_t NextSheet();
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine if the given sheet profile is calibrated.
+    ///
+    /// In case the index of the given print sheet profile is bigger than the
+    /// MAX_SHEETS method return false.
+    /// @param[in] index Index of the sheet profile
+    /// @return True when the profile is calibrated, False othewise.
+    static bool IsSheetCalibrated(uint32_t index);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Select the given print sheet profile as an active for the printer.
+    ///
+    /// In case the index of the given print sheet profile is bigger than the
+    /// MAX_SHEETS return false.
+    /// @param[in] index Index of the sheet profile
+    /// @return True when the profile can be selected, False othewise.
+    static bool SelectSheet(uint32_t index);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine the name of the current active print sheet profile.
+    ///
+    /// @param[out] buffer Buffer to store the print sheet profile
+    /// @param[in] length Size of the given buffer.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t ActiveSheetIndex();
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Reset the given print sheet profile to the uncalibrated state.
+    ///
+    /// In case the index of the given print sheet profile is bigger than the
+    /// MAX_SHEETS method return false.
+    /// @param[in] index Index of the sheet profile
+    /// @return True when the profile was reset, False othewise.
+    static bool ResetSheet(uint32_t index);
```

Comment:
Suggestion:
I consider documentation of this function too verbose.
What about:
    /// @brief Reset the given print sheet profile to the uncalibrated state.
    ///
    /// @param[in] index Index of the sheet profile
    /// @retval true success
    /// @retval false index out of range